libraries/abstractions/common_io/test/test_iot_gpio.c

```diff
@@ -97,6 +97,8 @@ TEST_GROUP( TEST_IOT_GPIO );
  */
 TEST_SETUP( TEST_IOT_GPIO )
 {
+    xtestIotGpioSemaphore = xSemaphoreCreateBinary();
```

Comment:
TEST_SETUP will be called before each test case. So you should delete the semaphore in TEST_TEAR_DOWN as well.